src/lib/sort/sort-header.ts

```diff
@@ -138,20 +139,26 @@ export class MatSortHeader extends _MatSortHeaderMixinBase
 
   constructor(public _intl: MatSortHeaderIntl,
               changeDetectorRef: ChangeDetectorRef,
+              /**
+               * @deprecated This sort is no longer used, use sortContainer instead.
+               * @breaking-change 8.0.0
+               */
```

Comment:
Actually, if we want to change this in version 8, we can just do it now since master is already targeting next major. You would just have to change this from "patch" to "major" (which is probably should be anyway)

Reply:
Sure - I'll go ahead and remove it from the constructor altogether